src/TagDependencyTrait.php

```diff
@@ -173,7 +173,7 @@ public function invalidateTags()
             ]
         );
 
-        if (!empty($this->cacheCompositeTagFields())) {
+        if ($this->cacheCompositeTagFields() != '') {
```

Comment:
`function cacheCompositeTagFields()` return array, not string. To check for not empty, you should compare with empty array